packages/@azure/servicebus/data-plane/test/testUtils.ts

```diff
@@ -193,19 +192,26 @@ export function getReceiverClient(
 }
 
 export async function purge(
-  recieverClient: QueueClient | SubscriptionClient,
+  receiverClient: QueueClient | SubscriptionClient,
   useSessions?: boolean
 ): Promise<void> {
-  const peekedMsgs = await recieverClient.peek();
-  if (!peekedMsgs.length) {
-    return;
-  }
+  let isEmpty = false;
 
   const receiver = useSessions
-    ? await recieverClient.getSessionReceiver()
-    : recieverClient.getReceiver();
+    ? await receiverClient.getSessionReceiver({ sessionId: "my-session" })
+    : receiverClient.getReceiver();
+
+  while (!isEmpty) {
+    const peekedMsgs = await receiverClient.peek();
```

Comment:
How about peeking for say 10 messages and then passing `peekedMsgs.length` to the receiveBatch() call?
We reduce the number of receiveBatch() calls being made.

Reply:
This wouldn't be ideal/precise because then we'll end up working with setting receiveBatch() timeout delays. (Which defaults to 60 sec)

Reply:
I was thinking of something like the below. Can you elaborate on your concern of timeout here?

```
  while (!isEmpty) {
     const peekedMsgs = await receiverClient.peek(10);
     if (peekedMsgs.length === 0) {
      isEmpty = true;
    } else {
      const msgs = await receiver.receiveBatch(peekedMsgs.length);
      for (let i = 0; i < msgs.length; i++) {
        await  msgs[i].complete();
      }
    }
  }
```